src/smc-webapp/project_store.coffee

```diff
@@ -890,6 +893,62 @@ class ProjectActions extends Actions
             else
                 return path
 
+    # this is called once by the project initialization
+    check_first_steps: =>
+        {NO_FIRST_STEPS_SENTINEL_FILE} = require('./project_files')
+        cmd = "test ! -e #{NO_FIRST_STEPS_SENTINEL_FILE} && test -e #{LIBRARY.first_steps}"
+        webapp_client.exec
+            project_id      : @project_id
+            command         : cmd
+            bash            : true
+            timeout         : 30
+            network_timeout : 120
+            err_on_exit     : false
+            path            : '.'
+            cb              : (err, output) =>
+                if output.exit_code == 0
+                    @setState(show_first_steps:true)
+                else
+                    @setState(show_first_steps:false)
+
+    copy_from_library: (opts) =>
+        opts = defaults opts,
+            src     : LIBRARY.first_steps
+            dest    : '.'
+
+        id = opts.id ? misc.uuid()
+        @set_activity(id:id, status:"Copying files from library ...")
+
+        webapp_client.exec
+            project_id      : @project_id
+            command         : 'rsync'
+            args            : ['-rltgoDxH'].concat(opts.src).concat([opts.dest])
+            timeout         : 120   # how long rsync runs on client
+            network_timeout : 120   # how long network call has until it must return something or get total error.
+            err_on_exit     : true
+            path            : '.'
+            cb              : (err, output) =>
+                (@_finish_exec(id))(err, output)
+                @open_file(path: 'first-steps/first-steps.tasks')
```

Comment:
1. Don't call open_file if there is err is true-ish.  Otherwise you'll just create a blank file.

2. Open something involving opts.dest not first-steps/first-steps.tasks, or eliminate dest.    Otherwise this will be buggy.